### PR TITLE
add vpc-endpoints to the workspace documentation

### PR DESCRIPTION
**Pull request by nkvuong**
Add VPC endpoints to the E2 Terraform deployment to line up with existing [docs](https://docs.databricks.com/administration-guide/cloud-configurations/aws/customer-managed-vpc.html#regional-endpoints-1), for more direct connections and reduced costs

### CODE REVIEW

File: docs/guides/aws-workspace.md [hunk 1]
```diff
@@ -112,9 +116,14 @@ The very first step is VPC creation with necessary firewall rules. Please consul
 ```hcl
 data "aws_availability_zones" "available" {}
 
+data "aws_security_group" "default" {
```
Comment (nfx) [inline docs/guides/aws-workspace.md:119]: this section doesn't need this data resource

File: docs/guides/aws-workspace.md [hunk 2]
```diff
@@ -149,6 +161,79 @@ resource "databricks_mws_networks" "this" {
...
+      test     = "StringNotEquals"
+      variable = "aws:sourceVpce"
+
+      values = [data.aws_vpc_endpoint_service.s3.id]
+    }
+  }
+}
+
+module "vpc_endpoints" {
+  source = "terraform-aws-modules/vpc/aws//modules/vpc-endpoints"
```
Comment (nfx) [inline docs/guides/aws-workspace.md:205]: remove double-slash

File: docs/guides/aws-workspace.md [hunk 3]
```diff
@@ -149,6 +161,79 @@ resource "databricks_mws_networks" "this" {
...
+  version = "3.2.0"
+
+  vpc_id             = module.vpc.vpc_id
+  security_group_ids = [data.aws_security_group.default.id]
```
Comment (nfx) [inline docs/guides/aws-workspace.md:209]: replace with `module.vpc.default_security_group_id` (https://registry.terraform.io/modules/terraform-aws-modules/vpc/aws/latest?tab=outputs)

File: docs/guides/aws-workspace.md [hunk 4]
```diff
@@ -149,6 +161,79 @@ resource "databricks_mws_networks" "this" {
...
+      service_type    = "Gateway"
+      route_table_ids = flatten([module.vpc.intra_route_table_ids, module.vpc.private_route_table_ids, module.vpc.public_route_table_ids])
+      policy          = data.aws_iam_policy_document.s3_endpoint_policy.json
+      tags            = { Name = "${local.prefix}-s3-vpc-endpoint" }
+    },
+    sts = {
+      service             = "sts"
+      private_dns_enabled = true
+      subnet_ids          = module.vpc.private_subnets
+      tags                = { Name = "${local.prefix}-sts-vpc-endpoint" }
```
Comment (nfx) [inline docs/guides/aws-workspace.md:223]: for consistency reasons, put Name on the new line:

e.g.

```
tags {
  Name = ...
}
```

**Review by nfx — COMMENTED** [review]
Small changes required

File: docs/guides/aws-workspace.md (hunk 2 above)
Comment (nkvuong) [reply docs/guides/aws-workspace.md:196]: double-slash is needed, as this is a sub-directory within a tf module

**Review by nfx — APPROVED** [review]

**Pull request merged**